app/post.go

```diff
@@ -27,9 +28,10 @@ import (
 )
 
 const (
-	PendingPostIDsCacheSize = 25000
-	PendingPostIDsCacheTTL  = 30 * time.Second
-	PageDefault             = 0
+	PendingPostIDsCacheSize         = 25000
+	PendingPostIDsCacheTTL          = 30 * time.Second
+	PageDefault                     = 0
+	VoiceMessageDurationErrorMargin = 3
```

Comment:
**short version:** is there's some small disparity between the client's mp3 duration and the server one we should allow the file to go through. the rule validation rule is: 

`success if mp3 duration < (admin rule + the above error margin)`

**long version:** About this new constant: I was thinking about a case where the client (webapp/mobile) would stop a recording automatically because the max duration is hit, but a few milliseconds would happen before this action happens. Having a strict validation on the duration could cause the client to send what it considers to be a valid voice message but the server would deny it. Adding a few extra seconds adds some flexibility while not being extremely penalizing. 
I thought 3secs offer plenty of flexibility without being too much. Even if the admin decides to set a pretty small limit, 30secs for example, allowing the receive messages up to 33seconds is not bad. it gets worst as the admin value goes down but I also think that anything below 30s will lead to a bad experience anyway

Reply:
I tend to agree here although I'd be very surprised if we weren't able to reach second precision on the client. Also we are aware of the limit on that side and could cut the recording short at any point before hitting that. A couple of seconds of tolerance makes sense though.